pkg/cmd/render/render.go

```diff
@@ -63,6 +84,41 @@ func (r *renderOpts) Validate() error {
 		return err
 	}
 
+	clusterInfrastructureFileLen := len(r.clusterInfrastructureFile)
+	cloudProviderConfigInputFileLen := len(r.cloudProviderConfigInputFile)
+	cloudProviderConfigOutputFileLine := len(r.cloudProviderConfigOutputFile)
+
+	// Validate all files are specified when specifying infrastructure and configmap files
+	if clusterInfrastructureFileLen != 0 || cloudProviderConfigInputFileLen != 0 || cloudProviderConfigOutputFileLine != 0 {
+		if !(clusterInfrastructureFileLen != 0 && cloudProviderConfigInputFileLen != 0 && cloudProviderConfigOutputFileLine != 0) {
+			return fmt.Errorf("clulster-infrastructure-file, cloud-provider-config-input-file, and cloud-provider-config-output-file must all be specified.")
+		}
```

Comment:
there is no requirement for input cloud provider file. see AWS, when using service endpoints there is no input configmap but there is a generated one with all the service overrides.

I think we need to take into account that.

Reply:
Still to-do.